assets/js/vue-apps/grant-filters.js

```diff
@@ -10,80 +11,145 @@ function init() {
+            // Populate data from global object (eg. to share server/client state)
+            let PGS = window._PAST_GRANTS_SEARCH;
+            let queryParams = PGS && PGS.queryParams ? PGS.queryParams : {};
+            let existingFacets = PGS && PGS.facets ? PGS.facets : {};
+            let existingSort = PGS && PGS.sort ? PGS.sort : {};
             let defaultFilters = {
                 amount: '',
                 year: '',
                 programme: '',
                 country: '',
                 localAuthority: ''
             };
-            let filters = Object.assign({}, defaultFilters);
             return {
-                facets,
                 defaultFilters,
-                filters
+                sort: Object.assign({}, existingSort),
+                facets: Object.assign({}, existingFacets),
+                filters: Object.assign({}, defaultFilters, queryParams),
+                isCalculating: false,
+                totalResults: PGS.totalResults || 0,
+                searchError: false
             };
         },
+        watch: {
+            // Watch for changes to filters then make AJAX call
+            filters: {
+                handler: function() {
+                    this.filterResults();
+                },
+                deep: true
+            }
+        },
         mounted: function() {
             // Enable inputs (they're disabled by default to avoid double inputs for non-JS users)
             $(this.$el)
                 .find('[disabled]')
                 .removeAttr('disabled');
-            // Populate the facets object
-            if (window._PAST_GRANTS_SEARCH && window._PAST_GRANTS_SEARCH.facets) {
-                this.facets = Object.assign({}, window._PAST_GRANTS_SEARCH.facets);
-            }
-            // Populate any existing query parameters
-            if (
-                window._PAST_GRANTS_SEARCH &&
-                window._PAST_GRANTS_SEARCH.queryParams &&
-                !!window._PAST_GRANTS_SEARCH.queryParams
-            ) {
-                this.filters = Object.assign({}, window._PAST_GRANTS_SEARCH.queryParams);
-            }
         },
         methods: {
-            sort: function(sortKey, direction) {
+            // Create the sort parameters
+            sortData: function(sortKey, currentDirection) {
+                // Flip reverse it
+                let direction = currentDirection === 'asc' ? 'desc' : 'asc';
+                this.sort = {
+                    type: sortKey,
+                    direction: direction
+                };
                 this.filters.sort = `${sortKey}|${direction}`;
                 this.filterResults();
             },
+
+            // Generate CSS classes for sort links
+            sortLinkClasses: function(sortKey) {
+                const type = this.sort.type;
+                const dir = this.sort.direction;
+                return {
+                    'sort-controls__item--active': type === sortKey,
```

Comment:
It might be a bit ugly having CSS classnames in JS, but on the other hand it looked quite ugly having chained JS logic in the HTML, so...

Reply:
In which case let's go for a "state" class like `is-active`, `is-descending`, `is-ascending`, rather than using component-specific classes.

Reply:
Done!